openupgrade_scripts/scripts/loyalty/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,815 @@
+    ("coupon.rule", "loyalty.rule"),
+]
+_tables_renames = [
+    ("coupon_program", "loyalty_program"),
+    ("coupon_reward", "loyalty_reward"),
+    ("coupon_coupon", "loyalty_card"),
+    ("coupon_rule", "loyalty_rule"),
+]
+_xmlids_renames = [
+    (
+        "loyalty.coupon_action",
+        "loyalty.loyalty_card_action",
+    ),
+    (
+        "loyalty.coupon_generate",
+        "loyalty.loyalty_generate_wizard_action",
+    ),
+    (
+        "loyalty.report_coupon_code",
+        "loyalty.report_loyalty_card",
+    ),
+    (
+        "loyalty.coupon_view_form",
+        "loyalty.loyalty_card_view_form",
+    ),
+    (
+        "loyalty.coupon_view_tree",
+        "loyalty.loyalty_card_view_tree",
+    ),
+    (
+        "loyalty.coupon_generate_view_form",
+        "loyalty.loyalty_generate_wizard_view_form",
+    ),
+    (
+        "loyalty.coupon_program_view_form_common",
+        "loyalty.loyalty_program_view_form",
+    ),
+    (
+        "loyalty.coupon_program_view_search",
+        "loyalty.loyalty_program_view_search",
+    ),
+    (
+        "loyalty.coupon_program_view_tree",
+        "loyalty.loyalty_program_view_tree",
+    ),
+    (
+        "sale_loyalty.mail_template_sale_coupon",
+        "loyalty.mail_template_loyalty_card",
+    ),
+    (
+        "gift_card.gift_card_product_50",
+        "loyalty.gift_card_product_50",
+    ),
+]
+
+
+def update_loyalty_program_data(env):
+    # Fill date_to fill: This data takes the values of the following fields:
+    # - validity_duration (coupon.program): in case of coupon programs that generate
+    # coupons with a validity time, it is calculated by taking the creation date of
+    # the program to which the duration defined in the field is added to obtain the
+    # date.
+    # - rule_date_to (coupon.rule): in the case of promotional programs, the date of
+    # rule_date_to is taken.
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_program
+        ADD COLUMN IF NOT EXISTS date_to DATE
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET date_to = CASE
+            WHEN validity_duration != 0 AND date_to IS NULL
+                THEN create_date + validity_duration * interval '1 day'
+            WHEN loyalty_rule.rule_date_to IS NOT NULL AND date_to IS NULL
+                THEN loyalty_rule.rule_date_to
+        END
+        FROM loyalty_rule
+        WHERE loyalty_program.validity_duration != 0
+        AND loyalty_program.rule_id = loyalty_rule.id
+        """,
+    )
+    # Set limit usage values
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_program
+        ADD COLUMN IF NOT EXISTS limit_usage BOOLEAN
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET limit_usage = TRUE
+        WHERE max_usage IS NOT NULL
```

Comment:
Did you rename `maximum_use_number`?

Reply:
Yes, here: https://github.com/OCA/OpenUpgrade/blob/824798f4dc539f115a686872e3342dee89a4c7f4/openupgrade_scripts/scripts/loyalty/16.0.1.0/pre-migration.py#L8

But with misspelling, I correct it :S